locking/locks.go

```diff
@@ -225,33 +249,101 @@ func (c *Client) lockIdFromPath(path string) (string, error) {
 	}
 }
 
+// We want to use a single cache file for integrity, but to make it easy to
+// list all locks, prefix the id->path map in a way we can identify (something
+// that won't be in a path)
+const idKeyPrefix = "*id*://"
+
+func (c *Client) encodeIdKey(id string) string {
+	// Safety against accidents
+	if !c.isIdKey(id) {
+		return idKeyPrefix + id
+	}
+	return id
+}
+func (c *Client) decodeIdKey(key string) string {
+	// Safety against accidents
+	if c.isIdKey(key) {
+		return key[len(idKeyPrefix):]
+	}
+	return key
+}
+func (c *Client) isIdKey(key string) bool {
+	return strings.HasPrefix(key, idKeyPrefix)
+}
+
 // Cache a successful lock for faster local lookup later
-func (c *Client) cacheLock(filePath, id string) error {
-	// TODO
+func (c *Client) cacheLock(l Lock) error {
+	// Store reference in both directions
+	// Path -> Lock
+	c.cache.Set(l.Path, &l)
+	// EncodedId -> Lock (encoded so we can easily identify)
+	c.cache.Set(c.encodeIdKey(l.Id), &l)
 	return nil
 }
 
 // Remove a cached lock by path becuase it's been relinquished
-func (c *Client) cacheUnlock(filePath string) error {
-	// TODO
+func (c *Client) cacheUnlockByPath(filePath string) error {
+	ilock := c.cache.Get(filePath)
+	if ilock != nil {
+		lock := ilock.(*Lock)
+		c.cache.Remove(lock.Path)
+		// Id as key is encoded
+		c.cache.Remove(c.encodeIdKey(lock.Id))
+	}
 	return nil
 }
 
-// Remove a cached lock by id becuase it's been relinquished
+// Remove a cached lock by id because it's been relinquished
 func (c *Client) cacheUnlockById(id string) error {
-	// TODO
+	// Id as key is encoded
+	idkey := c.encodeIdKey(id)
+	ilock := c.cache.Get(idkey)
+	if ilock != nil {
+		lock := ilock.(*Lock)
+		c.cache.Remove(idkey)
+		c.cache.Remove(lock.Path)
+	}
 	return nil
```

Comment:
Do you think we should return an error if we couldn't find the lock? I think a sentinel value here would be fine:

```go
package locking

var ErrMissingLock error = errors.New("locking: could not find lock")
```